Remove tint log colouriser dependency

Tint https://github.com/lmittmann/tint when updated from 0.3.4 (our original version) to the latest of 1.1.2:

- https://github.com/lmittmann/tint

broke `go build` when updating all dependencies via: `go get -u ./...` with a follow-up tidy of: `go mod tidy`.

```
 go build

# github.com/rockstor/rockon-validator
./main.go:104:10: cannot use logLevel (variable of type *"golang.org/x/exp/slog".LevelVar) as "log/slog".Leveler value in struct literal: *"golang.org/x/exp/slog".LevelVar does not implement "log/slog".Leveler (wrong type for method Level)
                have Level() "golang.org/x/exp/slog".Level
                want Level() "log/slog".Level
./main.go:105:16: cannot use func(groups []string, attr slog.Attr) slog.Attr {…} (value of type func(groups []string, attr "golang.org/x/exp/slog".Attr) "golang.org/x/exp/slog".Attr) as func(groups []string, attr "log/slog".Attr) "log/slog".Attr value in struct literal
./main.go:113:20: cannot use logHandler (variable of interface type "log/slog".Handler) as "golang.org/x/exp/slog".Handler value in argument to slog.New: "log/slog".Handler does not implement "golang.org/x/exp/slog".Handler (wrong type for method Enabled)
                have Enabled(context.Context, "log/slog".Level) bool
                want Enabled(context.Context, "golang.org/x/exp/slog".Level) bool
```

This is unwanted behaviour for a log colouriser so moving to remove completely: in line with the directive established under:

- https://github.com/rockstor/rockon-validator/issues/16

---

Re-introduction using all new dependencies is welcome, if anyone is interested in contributing, but is not considered a priority.

--- a/README.md
+++ b/README.md
@@ -80,53 +80,60 @@ Returns:
 Similarly, `--diff` produces a `diffutils` formated output re: existing and proposed file format:
 
 ```diff
---- a/files/bitcoind.json
-+++ b/files/bitcoind.json
-@@ -1,21 +1,25 @@
+./rockon-validator --diff ./temp/OpenSpeedTest.json 
+time=2026-01-05T13:42:05.137Z level=WARN msg="Name mismatch:" index="SpeedTest by OpenSpeedTest" expected="speedtest by openspeedtest" file=OpenSpeedTest.json
+time=2026-01-05T13:42:05.137Z level=WARN msg="(if --write) Removing and adding expected entry."
+--- a/./temp/OpenSpeedTest.json
++++ b/./temp/OpenSpeedTest.json
+@@ -1,25 +1,21 @@
  {
-     "Bitcoin": {
-+        "description": "Bitcoin full node. <p>Based on a custom docker image: <a href='https://hub.docker.com/r/kylemanna/bitcoind' target='_blank'>https://hub.docker.com/r/kylemanna/bitcoind</a>, available for amd64 architecture only.</p>",
-+        "version": "1.1",
-+        "website": "https://bitcoin.org/en/full-node",
-+        "volume_add_support": true,
-         "containers": {
-             "bitcoind": {
-                 "image": "kylemanna/bitcoind",
-                 "launch_order": 1,
-                 "ports": {
--                    "8333": {
--                        "description": "Listening port",
--                        "host_default": 28333,
--                        "label": "Port for incoming connections",
--                        "protocol": "tcp"
--                    },
-                     "8332": {
-                         "description": "JSONRPC port",
--                        "host_default": 28332,
-                         "label": "The JSONRPC server allows to query and control the server remotely",
-+                        "host_default": 28332,
-                         "protocol": "tcp"
-+                    },
-+                    "8333": {
-+                        "description": "Listening port",
-+                        "label": "Port for incoming connections",
-+                        "host_default": 28333,
-+                        "protocol": "tcp"
-                     }
-                 },
-                 "volumes": {
-@@ -26,10 +21,6 @@
-                     }
-                 }
-             }
--        },
--        "description": "Bitcoin full node. <p>Based on a custom docker image: <a href='https://hub.docker.com/r/kylemanna/bitcoind' target='_blank'>https://hub.docker.com/r/kylemanna/bitcoind</a>, available for amd64 architecture only.</p>",
--        "volume_add_support": true,
--        "website": "https://bitcoin.org/en/full-node",
--        "version": "1.1"
+   "SpeedTest by OpenSpeedTest": {
++    "description": "<p>SpeedTest by OpenSpeedTest™ is a free and open-source HTML5 network performance estimation tool. Written in vanilla Javascript it only uses built-in Web APIs like XMLHttpRequest (XHR), HTML, CSS, JS, &amp; SVG. No third-party frameworks or libraries are required.<p>Based on the official docker image: <a href='https://hub.docker.com/r/openspeedtest/latest' target='_blank'> https://hub.docker.com/r/openspeedtest/latest/</a>, available for amd64 and arm64 architecture.</p>",
++    "version": "1.0",
++    "website": "https://openspeedtest.com/",
+     "containers": {
+       "speedtest": {
+         "image": "openspeedtest/latest",
+-               "launch_order": 1,
+-               "ports": {
+-                       "3000": {
+-                               "description": "SpeedTest http WebUI port. Suggested Default: 3000",
+-                               "host_default": 3000,
+-                               "label": "http WebUI port",
+-                               "ui": true
+-                       }
+-               }
++        "launch_order": 1,
++        "ports": {
++          "3000": {
++            "description": "SpeedTest http WebUI port. Suggested Default: 3000",
++            "label": "http WebUI port",
++            "host_default": 3000,
++            "ui": true
++          }
 +        }
-     }
+       }
+-    },
+-    "description": "<p>SpeedTest by OpenSpeedTest™ is a free and open-source HTML5 network performance estimation tool. Written in vanilla Javascript it only uses built-in Web APIs like XMLHttpRequest (XHR), HTML, CSS, JS, &amp; SVG. No third-party frameworks or libraries are required.<p>Based on the official docker image: <a href='https://hub.docker.com/r/openspeedtest/latest' target='_blank'> https://hub.docker.com/r/openspeedtest/latest/</a>, available for amd64 and arm64 architecture.</p>",
+-    "ui": {
+-      "https": false,
+-      "slug": ""
+-    },
+-    "website": "https://openspeedtest.com/",
+-    "version": "1.0"
++    }
+   }
  }
+# the following is abridged:
+--- a/temp/root.json
++++ b/temp/root.json
+@@ -1,89 +1,89 @@
+ {
+...
+-    "SpeedTest by OpenSpeedTest": "OpenSpeedTest.json",
+...
++  "speedtest by openspeedtest": "OpenSpeedTest.json"
+...
 ```
 
 N.B. `--write` **USE WITH CAUTION** re-writes the file in-place; assuming correct syntax but incorrect formatting.
@@ -148,7 +155,7 @@ Details more associated with development.
 ### GO download and install
 
 Alternatives to the `docker run` approach, and required for development purposes.
-Requires GO version 1.23 or later.
+Requires GO version 1.24 or later.
 - [Upstream install instructions](https://go.dev/doc/install)
 
 ### Go build

--- a/go.mod
+++ b/go.mod
@@ -1,9 +1,8 @@
 module github.com/rockstor/rockon-validator
 
-go 1.23
+go 1.24.0
 
 require (
 	github.com/hexops/gotextdiff v1.0.3
-	github.com/lmittmann/tint v0.3.4
-	golang.org/x/exp v0.0.0-20230713183714-613f0c0eb8a1
+	golang.org/x/exp v0.0.0-20251219203646-944ab1f22d93
 )

--- a/main.go
+++ b/main.go
@@ -10,13 +10,12 @@ import (
 	"path/filepath"
 	"slices"
 	"strings"
-
+	// https://pkg.go.dev/golang.org/x/exp/slog@v0.0.0-20251219203646-944ab1f22d93#Logger
 	"golang.org/x/exp/slog" // nee "log/slog"
 
 	"github.com/hexops/gotextdiff"
 	"github.com/hexops/gotextdiff/myers"
 	"github.com/hexops/gotextdiff/span"
-	"github.com/lmittmann/tint"
 
 	"github.com/rockstor/rockon-validator/model"
 )
@@ -74,45 +73,9 @@ func parseFileArgs() (filePaths []string) {
 		}
 		filePaths = append(filePaths, glob...)
 	}
-	// recurse subdirectories
-	//for i, f := range filePaths {
-	//	files, err := os.ReadDir(f)
-	//	if err != nil {
-	//		continue // What we got was not a directory, so we can leave it be
-	//	}
-	//
-	//	entries := []string{}
-	//	for _, e := range files {
-	//		if !e.IsDir() {
-	//			entries = append(entries, filepath.Join(f, e.Name()))
-	//		}
-	//	}
-	//	head := filePaths[:i]
-	//	if i == 0 {
-	//		head = []string{}
-	//	}
-	//	tail := filePaths[i+1:]
-	//	filePaths = append(head, entries...)
-	//	filePaths = append(filePaths, tail...)
-	//}
+
 	logger.Debug("paseFileArgs()", slog.Any("Return", filePaths))
 	return filePaths
-}
-
-func setupLogger(logLevel *slog.LevelVar) *slog.Logger {
-	logOpts := &tint.Options{
-		Level: logLevel,
-		ReplaceAttr: func(groups []string, attr slog.Attr) slog.Attr {
-			if attr.Key == slog.TimeKey && len(groups) == 0 {
-				return slog.Attr{}
-			}
-			return attr
-		},
-	}
-	logHandler := tint.NewHandler(os.Stderr, logOpts)
-	logger = slog.New(logHandler)
-	slog.SetDefault(logger)
-	return logger
 }
 
 func checkRootMap(rootMap map[string]string, filename string, rockon model.RockOn) {
@@ -148,9 +111,11 @@ func checkRootMap(rootMap map[string]string, filename string, rockon model.RockO
 }
 
 func main() {
-	logLevel := &slog.LevelVar{}
+	var logLevel = new(slog.LevelVar)
 	logLevel.Set(slog.LevelWarn)
-	logger = setupLogger(logLevel)
+	logHandler := slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: logLevel})
+	logger = slog.New(logHandler)
+	slog.SetDefault(logger)
 
 	flag.Usage = func() {
 		_, err := fmt.Fprint(os.Stderr, usage)

